validator/block_validator.go

```diff
@@ -202,7 +299,7 @@ func (v *BlockValidator) readLastBlockValidatedDbInfo(reorgingToBlock *types.Blo
 
 	if reorgingToBlock == nil {
 		expectedHash := v.blockchain.GetCanonicalHash(info.BlockNumber)
-		if expectedHash != info.BlockHash {
+		if expectedHash != info.BlockHash && (expectedHash != common.Hash{}) {
```

Comment:
Hmm.. In case a reorg happened while the validator wasn't enabled, it might make sense to roll back last validated here. That said, it's not very relevant to this PR.